### ISSUE TITLE
Provide a title for commit summaries

### ISSUE THREAD

**Issue by joshaber** [desktop/desktop#1483]
I wanna hover to see the whole summary.

<img width="297" alt="screen shot 2017-05-15 at 10 00 33 pm" src="https://cloud.githubusercontent.com/assets/13760/26086654/0257a5ec-39ba-11e7-97b0-ac939fb731b7.png">

**Comment by ahmed-taj** [comment]
@joshaber I would like to work on this

**Comment by niik** [comment]
@ahmed-taj :clap: Please do! Let us know if you would like any assistance.

**Comment by ahmed-taj** [comment]
Sorry for late response, but I took forever to make GitHub Desktop works on Ubuntu (the only OS I have).

I tried to follow the way github.com shows the whole summary. I think it adds a `title` attribute with the full commit message, right?

This is a screenshot of the result:
![full-summary](https://cloud.githubusercontent.com/assets/12673605/26250675/7a89d41c-3cb3-11e7-80d2-219db4a527a4.png)

I achieved that by only adding `title` to the `<span>` element in case that `token.kind` is text:

```javascript
// File: app/src/ui/lib/rich-text.tsx  

const elements = tokenizer.tokenize(str).map((token, index) => {
  switch (token.kind) {
    case TokenType.Emoji:
      return <img key={index} alt={token.text} title={token.text} className='emoji' src={token.path}/>
    case TokenType.Link:
      return <LinkButton key={index} uri={token.url} children={token.text} />
    case TokenType.Text:
      return <span key={index} title={token.text}>{token.text}</span> // <--- here
    default:
      return assertNever(token, 'Unknown token type: ${r.kind}')
  }
})
```
Ping @niik and @joshaber: Any comments or something I need to fix?

**Comment by niik** [comment]
I'm gonna reopen this as the fix provided in #1648 only adds title text for individual tokens in the commit list. Once you have a link or an emoji in there the individual parts of the commit summary have titles but we should really set a title element on the entire thing.

Illustrated below

![commit-title](https://cloud.githubusercontent.com/assets/634063/26265764/b29ba360-3ce2-11e7-9b68-517ba2b25847.gif)

In this case I would expect the title text when hovering over the commit message to be `hey :smile: having a good day?`

**Comment by ahmed-taj** [comment]
@niik oh, I will fix that
